modules/product/src/main/java/com/opengamma/strata/product/fxopt/ResolvedFxBinaryOption.java

```diff
@@ -0,0 +1,706 @@
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.currency.CurrencyPair;
+import com.opengamma.strata.product.ResolvedProduct;
+import com.opengamma.strata.product.common.LongShort;
+import com.opengamma.strata.product.common.PutCall;
+
+/**
+ * A binary FX option, resolved for pricing.
+ * <p>
+ * This is the resolved form of {@link FxBinaryOption} and is an input to the pricers.
+ * Applications will typically create a {@code ResolvedFxBinaryOption} from a {@code FxBinaryOption}
+ * using {@link FxBinaryOption#resolve(ReferenceData)}.
+ * <p>
+ * A {@code ResolvedFxBinaryOption} is bound to data that changes over time, such as holiday calendars.
+ * If the data changes, such as the addition of a new holiday, the resolved form will not be updated.
+ * Care must be taken when placing the resolved form in a cache or persistence layer.
+ */
+@BeanDefinition
+public final class ResolvedFxBinaryOption
+        implements ResolvedProduct, ImmutableBean, Serializable {
+
+    /**
+     * Whether the option is long or short.
+     * <p>
+     * At expiry, the long party will have the option to enter in this transaction;
+     * the short party will, at the option of the long party, potentially enter into the inverse transaction.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final LongShort longShort;
+    /**
+     * The expiry date-time of the option.
+     * <p>
+     * The option is European, and can only be exercised on the expiry date.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final ZonedDateTime expiry;
+    /**
+     * The reference FX Index of the option.
+     * <p>
+     * The reference FX Index is used to determine whether or not a payment event has occurred at the expiry
+     * of the option.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final FxIndex underlying;
+    /**
+     * The amount and currency of the option payment.
+     * <p>
+     * The payment amount that will be made to the long party in the specified currency, if at expiry, a payment
+     * event has occurred.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final AdjustablePayment paymentCurrencyAmount;
+    /**
+     * Whether the the option is a put or a call.
+     * <p>
+     * A call on the FX pair is not necessarily the same as a put on the inverse pair - unlike for vanilla
+     * options - since the payment currencies can be different in each case.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final PutCall putCall;
+    /**
+     * The strike FX rate for the binary option.
+     * <p>
+     * At expiry, the strike rate of the option will be compared to the reference FX Index to determine whether or not
+     * a payment event has occurred.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final double strike;
```

Comment:
Is it assumed that this strike value is based on `CurrencyPair` in `FxIndex`? If so, add the description to javadoc. IF not, use `FxRate`.